cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -621,6 +616,54 @@ mkConsensusTracers mbEKGDirect trSel verb tr nodeKern fStats = do
      when (isRollForward ev) $
        sendEKGDirectCounter ekgDirect "cardano.node.metrics.served.header.counter.int"
 
+traceBlockFetchServerMetrics
+  :: forall blk. ()
+  => Maybe EKGDirect
+  -> STM.TVar Int64
+  -> STM.TVar Int64
+  -> STM.TVar SlotNo
+  -> Tracer IO (TraceBlockFetchServerEvent blk)
+  -> Tracer IO (TraceBlockFetchServerEvent blk)
+traceBlockFetchServerMetrics Nothing _ _ _ tracer = tracer
+traceBlockFetchServerMetrics (Just ekgDirect) tBlocksServed tLocalUp tMaxSlotNo tracer = Tracer bsTracer
+
+  where
+    bsTracer :: TraceBlockFetchServerEvent blk -> IO ()
+    bsTracer e@(TraceBlockFetchServerSendBlock p) = do
+      traceWith tracer e
+      let mSlotNo = case pointSlot p of
+                         Origin -> Nothing
+                         At slotNo -> Just slotNo
+
```

Comment:
It doesn't seem necessary to transform one maybe type into another one.  A few lines below we pattern match on `mSlotNo`, equally well we can just pattern match on `pointSlot p`.